libraries/state_history/log.cpp

```diff
@@ -294,27 +294,47 @@ state_history_traces_log::state_history_traces_log(const state_history_config& c
 
 chain::bytes state_history_traces_log::get_log_entry(block_num_type block_num) {
 
-   auto get_traces_bin = [](auto& ds, uint32_t version) {
-      if (version == 0) {
-         return state_history::zlib_decompress(ds);
-      }
-      else {
-         std::vector<state_history::transaction_trace> traces;
-         state_history::trace_converter::unpack(ds, traces);
-         return fc::raw::pack(traces);
+   auto get_traces_bin = [block_num](auto& ds, uint32_t version, std::size_t size) {
+      auto start_pos = ds.tellp();
+      try {
+         if (version == 0) {
+            return state_history::zlib_decompress(ds);
+         }
+         else {
```

Comment:
Should we checking version 1, and throw out exception for other versions?

Reply:
Version has been checked in line 56 and I don't want to assume that the code here definitely won't work for future versions.